### PR TITLE
[decoupled-execution] buffer manager main loop logic (w/o message retry)

### PR DESCRIPTION
**Pull request by yuxiamit**
also removed the response from ther persisting phase (assuming state_computer's commit function handles the error by panics)

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by zekun000** [comment]
I think it'd be better to move those send request out to main loop to make it more explicit how the flow works. e.g.
```
if Some(item) = self.process_ordered_block(..).await {
  send_execution_request
}

if Some(item) = self.process_execution_response(..).await {
  send_signing_request
}
```

File: consensus/src/experimental/buffer_manager.rs [hunk 1]
```diff
@@ -418,10 +419,7 @@ impl StateManager {
         Ok(())
     }
 
-    async fn process_signing_response(
-        &mut self,
-        signing_resp: SigningResponse,
-    ) -> anyhow::Result<()> {
+    async fn process_signing_resp(&mut self, signing_resp: SigningResponse) -> anyhow::Result<()> {
```
[the remark this answers is not in the document]
Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:422]: resolved

File: consensus/src/experimental/buffer_manager.rs [hunk 2]
```diff
@@ -502,8 +500,37 @@ impl StateManager {
...
+        while !self.end_epoch {
+            // TODO: retry sending commit votes periodically
 
+            // process new messages
+            if let Err(e) = tokio::select! {
+                Some(blocks) = self.block_rx.next() => {
+                    self.process_ordered_blocks(blocks).await
+                }
+                Some(reset_event) = self.sync_rx.next() => {
+                    self.process_sync_req(reset_event).await
```
[the remark this answers is not in the document]
Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:515]: resolved

**Comment by yuxiamit** [comment]
> I think it'd be better to move those send request out to main loop to make it more explicit how the flow works. e.g.
> 
> ```
> if Some(item) = self.process_ordered_block(..).await {
>   send_execution_request
> }
> 
> if Some(item) = self.process_execution_response(..).await {
>   send_signing_request
> }
> ```

I think it might complicate the code because a `process` function might send two types of messages (e.g., processing execution response will result in sending a signing request (if succeeded) or an execution request (if not)).

I added some comments at the beginning of the function.

**Review by zekun000 — DISMISSED** [review]

**Pull request merged**